packages/definitions/dist/fhir/r4/profiles-medplum.json

```diff
@@ -3849,5 +3849,356 @@
+              "max" : "1"
+            }
+          },
+          {
+            "id" : "Agent.channel",
+            "path" : "Agent.channel",
+            "short" : "The channel on which to report matches to the criteria",
+            "definition" : "Details where to send notifications when resources are received that meet the criteria.",
+            "min" : 1,
+            "max" : "*",
+            "base" : {
+              "path" : "Agent.channel",
+              "min" : 1,
+              "max" : "*"
+            },
+            "type" : [{
+              "code" : "BackboneElement"
+            }]
+          },
+          {
+            "id" : "Agent.channel.name",
+            "path" : "Agent.channel.name",
+            "short" : "The human readable friendly name of the channel.",
+            "definition" : "The human readable friendly name of the channel.",
+            "min" : 1,
+            "max" : "1",
+            "base" : {
+              "path" : "Agent.channel.name",
+              "min" : 1,
+              "max" : "1"
+            },
+            "type" : [{
+              "code" : "string"
+            }]
+          },
+          {
+            "id" : "Agent.channel.status",
+            "path" : "Agent.status",
+            "short" : "active | off | error",
+            "definition" : "The status of the channel.",
+            "min" : 1,
+            "max" : "1",
+            "base" : {
+              "path" : "Agent.channel.status",
+              "min" : 1,
+              "max" : "1"
+            },
+            "type" : [{
+              "code" : "code"
+            }],
+            "binding" : {
+              "strength" : "required",
+              "valueSet" : "https://medplum.com/fhir/ValueSet/agent-status|4.0.1"
+            }
+          },
+          {
+            "id" : "Agent.channel.device",
+            "path" : "Agent.channel.device",
+            "definition" : "Optional device resource representing the device connected to the channel.",
+            "min" : 0,
+            "max" : "1",
+            "type" : [{
+              "code" : "Reference",
+              "targetProfile" : ["http://hl7.org/fhir/StructureDefinition/Device"]
+            }],
+            "base" : {
+              "path" : "Agent.channel.device",
+              "min" : 0,
+              "max" : "1"
+            }
+          },
+          {
+            "id" : "Agent.channel.target",
+            "path" : "Agent.channel.target",
+            "definition" : "The target resource where channel messages will be delivered.",
+            "min" : 1,
+            "max" : "1",
+            "type" : [{
+              "code" : "Reference",
+              "targetProfile" : ["https://medplum.com/fhir/StructureDefinition/Bot"]
+            }],
+            "base" : {
+              "path" : "Agent.channel.target",
+              "min" : 1,
+              "max" : "1"
+            }
+          },
+          {
+            "id" : "Agent.channel.bindHost",
```

Comment:
Might want to consider using the `Endpoint` resource for some of these properties related to the connection. Leverages the spec, and has terminology bindings for mimetype, etc.

Haven't done an in-depth analysis of the pros and cons, but having the Endpoint as a separate resource might make it easier to update across multiple agents


example

```json
{
      "resourceType" : "Endpoint",
      "id" : "77",
      "text" : {
        "status" : "generated",
        "div" : "<div xmlns=\"http://www.w3.org/1999/xhtml\">\n              v2 referral Requests, REF-I12<br/>\n              Address: mllp://127.0.0.1:4001</div>"
      },
      "status" : "active",
      "connectionType" : [{
        "coding" : [{
          "system" : "http://terminology.hl7.org/CodeSystem/endpoint-connection-type",
          "code" : "hl7v2-mllp"
        }]
      }],
      "name" : "v2 referral Requests",
      "managingOrganization" : {
        "display" : "Epic demo organization"
      },
      "payload" : [{
        "type" : [{
          "coding" : [{
            "code" : "REF-I12"
          }]
        }],
        "mimeType" : ["x-application/hl7-v2+er7"]
      }],
      "address" : "mllp://127.0.0.1:4001"
    }
```

Reply:
If you don't think this will be too much of a training/support burden, then I support this as the "right" data model